core/src/main/java/cucumber/runner/SynchronizedEventBus.java

```diff
@@ -1,41 +1,39 @@
 package cucumber.runner;
 
 import cucumber.api.event.Event;
-import cucumber.api.event.EventHandler;
 
-import java.util.Collections;
+import java.util.Collection;
 
-public class SynchronizedEventBus implements EventBus {
+public final class SynchronizedEventBus extends AbstractEventBus {
 
     private final EventBus delegate;
 
-    public SynchronizedEventBus(final EventBus delegate) {
+    public static SynchronizedEventBus synchronize(EventBus eventBus) {
+        if (eventBus instanceof SynchronizedEventBus) {
+            return (SynchronizedEventBus) eventBus;
+        }
+
+        return new SynchronizedEventBus(eventBus);
```

Comment:
deleted earlier comment didn't spot that this method was only being called the once